agent/src/main.cpp

```diff
@@ -177,9 +188,13 @@ int main(int argc, char **argv)
      * over the static configuration.
      */
     std::map<std::string, std::string> stringOptions;
-    if (servicemanifest != "") {
+    if (manifestDir != "") {
```

Comment:
Descriptive variable name > short variable name. I would go for serviceManifestDir and defaultServiceManifestDir, even though they are long. The directories and their names and functions are complicated enough as it is.